distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -731,71 +734,113 @@ type Table
-                    problem_builder.append (Column_Indexes_Out_Of_Range.Error invalid_indices.to_vector)
-                columns.map_with_index find_datatype
-
-        new_columns = columns.zip expected_types column-> expected_type-> case expected_type of
-            Nothing -> column
-            _ ->
-                parser = if expected_type == Auto then value_formatter.make_auto_parser else
-                    value_formatter.make_datatype_parser expected_type
-                storage = column.java_column.getStorage
-                new_storage_and_problems = parser.parseColumn column.name storage
-                new_storage = new_storage_and_problems.value
-                problems = Vector.from_polyglot_array new_storage_and_problems.problems . map (Parse_Values_Helper.translate_parsing_problem expected_type)
-                problems.each problem_builder.append
-                Column.Value (Java_Column.new column.name new_storage)
-
-        ## TODO [RW] this case of is a workaround for wrong dataflow handling on arrays, it can be removed once the PR fixing it is merged, the relevant PR is:
-           https://github.com/enso-org/enso/pull/3400
-        result = Table.new new_columns
-        on_problems.attach_problems_after result problem_builder.to_vector
+
+       Arguments:
+       - columns: The columns to parse. If not specified, all text columns
+         will be parsed.
+       - type: The type to parse the columns to. Defaults to `Auto` meaning that
+         the type will be inferred from the data. If all values fit one of the
+         more specific types, that type is selected. Otherwise the result
+         remains unchanged (a Text column). The types are tried in the following
+         order: Integer, Decimal, Date_Time, Date, Time_Of_Day, Boolean.
+       - format: The formatting settings to use when parsing the columns.
+         Defaults to `Nothing` meaning that the default formatting settings will
+         be used. These settings can be overridden by providing a custom
+         `Data_Formatter`. Alternatively, when type is `Date`, `Date_Time`,
+         `Time_Of_Day` or `Boolean`, a text format can be used to specify the
+         format for the chosen datatype. For dates, it is the usual date format
+         notation, and for booleans it should be two values that represent true
+         and false, separated by a `|`.
+       - error_on_missing_columns: Specifies if a missing input column should
+         result in an error regardless of the `on_problems` settings. Defaults
+         to `True`.
+       - on_problems: Specifies how to handle if a problem occurs, raising as a
+         warning by default.
+
+       ! Error Conditions
+
+         - If a column in `columns` is not in the input table, a
+           `Missing_Input_Columns` is raised as an error, unless
+           `error_on_missing_columns` is set to `False`, in which case the
+           problem is reported according to the `on_problems` setting.
```

Comment:
This feels like some duplication of the the argument definition. 
```suggestion
         - If a column in `columns` is not in the input table, a
           `Missing_Input_Columns` is raised as an error or problem
           following the `error_on_missing_columns` rules.
```

Reply:
I've been using this style before, so it is just consistent with other places where we list these `Error Conditions`, I almost copy-pasted this part. So I guess if we want to change it, we probably should revisit these ones too. I just wanted a clear description of when this is reported as error and when not, in one place. How else do you think this should be phrased?